frontend/src/lib/components/PropertyFilters/components/taxonomicPropertyFilterLogic.ts

```diff
@@ -92,6 +88,13 @@ export const taxonomicPropertyFilterLogic = kea<taxonomicPropertyFilterLogicType
                     taxonomicGroup,
                     values.describeProperty
                 )
+
+                // Add cohort name if this is a cohort filter
+                if (propertyType === 'cohort' && item?.name) {
+                    const cohortFilter = filter as CohortPropertyFilter
+                    cohortFilter.cohort_name = item.name
+                }
```

Comment:
this makes it so that the cohort name is attached to the taxonomic property filter before we've actually saved the filter.